src/App.js

```diff
@@ -2,355 +2,76 @@ import React from "react";
-          return this.setCharAt(result, result.length - 1, '*')
-        }
-        if ("*".includes(button) && result[result.length -1] === "+") {
-          return this.setCharAt(result, result.length - 1, '*')
-        }
-      }
-    }
-    if ("/*+-(".includes(result[result.length - 2])) {
-      console.log('inside of line 234')
-      if ("-+".includes(result[result.length - 1])) {
-        if ("0123456789(".includes(button)) {
-          return result + button;
-        }
-        if (".".includes(button)) {
-          return result + "0.";
-        }
+      if (button.key === "-" && hasLastChar("-")) {
+        currentBtn = ""; //no double negatives allowed
       }
-    }
-    if ((result[result.length - 1] || "").includes(".")) {
-      console.log(result[result.length - 1]);
-      if ("0123456789".includes(button)) {
-        return result + button;
-      }
-    }
 
-       //🪲 72.2 + doesn't work FIXED
-    if ((result || "").includes('.')) {
-
-      if ("+-*/0123456789".includes(button)) {
-        return result + button;
+      if ((button.key === "*" || button.key === "/") && result === "") {
+        tempResult = 0; //insert zero in front of multiply or divide
       }
-    }
 
-    // 🪲 72(7+2 returns "not yet coded"
-
-
-    // operator + operator -> replace the first operator
-
-    // "" - - -> is not possible
-    //return "not yet coded";
-
-    return result;
-  };
-
-  closeParens = (result) => {
-    var numberOfOpenP = (result.match(/\(/g) || []).length;
-    var numberOfCloseP = (result.match(/\)/g) || []).length;
-    //this solution works but only if there is one parenthesis missing,
-    //instead, account for all parens missing
-
-    while (numberOfOpenP > numberOfCloseP) {
-      result = result + ")";
-      numberOfOpenP--;
+      this.setState({
+        result: tempResult + currentBtn,
+      });
     }
-    return result
-  };
-
-  getLastChar = (from) => {
-    return from.slice(-1);
-  };
-
-
-  roundedResult = (expression) => {
-    const digitAfterComma = 13;
-    round(evaluate(expression), digitAfterComma);
-  };
 
-  addMultiplier = (expression) => {
-    //Find all pattern Digit + (
-    //expression ="1(2+6)+23(6+9)+(2+3)9"
-
-    const digitAndP = /([0123456789])(\()/g;
-    const pAndDigit = /(\))([0123456789])/g;
-    // )( -> )*(
-    const pAndp = /(\))(\()/g;
-
-    //return the value
-    return expression
-      .replace(digitAndP, "$1*$2")
-      .replace(pAndDigit, "$1*$2")
-      .replace(pAndp, "$1*$2");
-
-
-  };
-
-
-
-  calculate = () => {
-    const { result, operate, originalLastNum } = this.state;
-    let finalResult = 0;
-    var tempResult = result;
-
-    // Special case with expression ending with operator, after CE use
-    if ("-+*/".includes(this.getLastChar(result))) {
-      tempResult = result.slice(0, -1);
+    if (lastEquation !== "" && button.key !== "=") {
+      //start new equation
+      this.setState({ result: button.key });
     }
```

Comment:
It does seem necessary because taking it out then adds the new number that is entered to the end of the last result rather than starting a new equation. Are you suggesting I can add code to isDisplayable to replace this? Because to answer your question, it is not currently handled by it.

Reply:
@grimxreaper What would we need to change in the isDisplayable section to remove it?

Reply:
If you figure out what would need to happen, go forward with the change.